rules/linux/initial_access_login_failures.toml

```
@@ -0,0 +1,51 @@
+[metadata]
+creation_date = "2020/07/08"
+ecs_version = ["1.5.0"]
+maturity = "development"
+updated_date = "2020/07/08"
+
+[rule]
+author = ["Elastic"]
+description = "Identifies that the maximum number of failed login attempts has been reached for a user."
+index = ["auditbeat-*"]
+interval = "10m"
+language = "kuery"
+license = "Elastic License"
+name = "Auditd Max Failed Login Attempts"
+references = ["https://github.com/linux-pam/linux-pam/blob/0adbaeb273da1d45213134aa271e95987103281c/modules/pam_faillock/pam_faillock.c#L574"]
+risk_score = 47
+rule_id = "fb9937ce-7e21-46bf-831d-1ad96eac674d"
+severity = "medium"
+tags = ["Elastic","Linux"]
+type = "query"
+
+query = '''
+event.module:auditd and auditd.message_type:anom_login_failures
```

Actually, I would change this even further to avoid non-ECS fields altogether

```suggestion
event.module:auditd and event.category:anomoly and event.action:failed-log-in-too-many-times-to
```

---

`event.module` is ECS, so it's okay to leave that

---

so I'd be fine if we really wanted to trigger on the `event.action` -- two things about the `event.category`:

1. It's misspelled and we treated that as a bug, so it's now `anomaly` in 7.9+
2. `anomaly` isn't a valid ECS category, so I have an existing PR in the underlying library that interfaces with the audit framework that adds proper ECS categorization to these, in likely 7.10+ they'll be `[anomaly, intrusion_detection, authentication]` and in 8 we're planning on making the breaking change and dropping `anomaly`

I figured though that the most stable of all of these for now is actually `auditd.message_type` as that's literally the mapped constant value that shows up in auditd log messages (i.e. `type=SERVICE_START msg=audit(1594133953.253:89): ...` will have an `auditd.message_type: service_start` value.

---

I noticed the spelling error and thought it was odd 🙂 .

if `auditd.message_type` is populated by the `type` then what is populating `event.action`, since they differ slightly?

We can use non-ECS fields but try to minimize, and the only way I see a promotion rule working here is if we can group all these `ANOM_*` events by the `event.category` or other field.

---

The `event.action` comes from our beefy normalization file here: https://github.com/elastic/go-libaudit/blob/2430e8c92f4426976ddb9068842413e4650f45b8/aucoalesce/normalizations.yaml#L1475-L1477

There's a bunch of code that parses the messages coming from the kernel over netlink then takes a look at the message type and normalizes/enriches the event based off of two things:

1. The message type itself
2. If the message type is an `AUDIT_SYSCALL` message, then the syscall (so we can categorize [`accept`](https://github.com/elastic/go-libaudit/blob/2430e8c92f4426976ddb9068842413e4650f45b8/aucoalesce/normalizations.yaml#L356) calls differently than say [`unlink`](https://github.com/elastic/go-libaudit/blob/2430e8c92f4426976ddb9068842413e4650f45b8/aucoalesce/normalizations.yaml#L310) calls)

Since these 4 events aren't based off of syscalls we just enrich them based off of the message type

---

But yeah, main difference between the `message_type` stuff v. `action` -- `action` is just some verbage we filled in that is somewhat arbitrary with an attempt to be descriptive about what's happening (and hence less stable). `message_type` maps 1-1 with the actual audit framework event type.

---

@andrewstucki @brokensound77 can you comment on the status of this PR?